src/hotspot/share/jfr/periodic/jfrPeriodic.cpp

```diff
@@ -644,3 +643,14 @@ TRACE_REQUEST_FUNC(ShenandoahHeapRegionInformation) {
   }
 #endif
 }
+
+TRACE_REQUEST_FUNC(HeapUsageSummary) {
+  EventHeapUsageSummary event;
+  if (event.should_commit()) {
```

Comment:
I believe the `should_commit` check is not needed; the period check is handle by the caller.

Reply:
👍